app/webpacker/controllers/mixins/useOpenAndCloseAsAModal.js

```diff
@@ -12,8 +12,8 @@ export const useOpenAndCloseAsAModal = (controller) => {
     }.bind(controller),
 
     close: function (_event, remove = false) {
-      // Only execute close if there is an open modal
-      if (!document.querySelector("body").classList.contains('modal-open')) return;
+      // Only execute close if the current modal is open
+      if (!this.modalTarget.classList.contains('in')) return;
```

Comment:
This `close` function is executed whenever the `modal:close` event is triggered.
On the Orders page, we have 3 confirmation modals e.g. 
- Resend Confirmation
- Send Invoice Confirmation
- Cancel Order Confirmation
Consider the scenario where the _Cancel Order Confirmation_ Modal is confirmed, then the `modal:close` event triggers. This `close` function gets triggered 3 times for each confirmation modal in sequential order (_Resend Confirmation Modal -> Send Invoice Confirmation -> Cancel Order Confirmation_) as each is listening for this event.
On the first `close` execution (_for Resend Confirmation Modal_), the `modal-open` class is removed from the body as per [Line#20](https://github.com/openfoodfoundation/openfoodnetwork/pull/13004/files#diff-b40716814842983e1b31e36b0292b142d9b08612d2af5ca54b7087a5408aee38R20).
So, on further executions (_for Send Invoice Confirmation and Cancel Order Confirmation_), the [existing guard condition](https://github.com/openfoodfoundation/openfoodnetwork/pull/13004/files#diff-b40716814842983e1b31e36b0292b142d9b08612d2af5ca54b7087a5408aee38L16) would always be `true`, causing the method to return without any close operation. Further to this point, no other confirmation modal will be closed on the page. 
So, updating the guard condition such that the `close` function only triggers if the current modal is opened.
I hope this explanation makes sense, please let me know if I'm missing something.